packages/svelte2tsx/src/htmlxtojsx_v2/nodes/Binding.ts

```diff
@@ -94,6 +94,11 @@ export function handleBinding(
         return;
     }
 
+    const expressionStr = str.original.substring(attr.expression.start, getEnd(attr.expression));
```

Comment:
```suggestion
    // add reassignment to force TS to widen the type of the declaration (in case it's never reassigned anywhere else)
    const expressionStr = str.original.substring(attr.expression.start, getEnd(attr.expression));
```